src/general/MainMenu.cs

```diff
@@ -521,6 +524,22 @@ private void SetBackgroundScene(string path)
         });
     }
 
+    private Control? GetCurrentMenu()
+    {
+        if (menuArray == null)
+            throw new InvalidOperationException("Main menu has not been initialized");
+        if (menuArray.Count <= 0)
+            throw new InvalidOperationException("Main menu has no menus");
+
+        foreach (Control menu in menuArray)
```

Comment:
Can't we just get a parent from first menu and then get a child on index CurrentMenuIndex to avoid this loop (and return null if CurrentMenuIndex is uint.MaxValue)? Or even better, save a reference to the parent (Menus node).

Reply:
I inspired myself too much in the code that handled menus and forgot that GetChild was a thing. :D
I changed the code as suggested.

Reply:
I think that you should also add the check for uint.MaxValue because GetChild would throw an exception in that case, but it should return null, I think. (that's my understanding of it, I am not a part of thrive dev team)

Reply:
Could also have bypassed this method altogether and set the visibility of the menu's root accordingly, but better to keep this in.